tempesta_fw/http_parser.c

```diff
@@ -3072,6 +3072,157 @@ __parse_uri_mark(TfwHttpReq *req, unsigned char *data, size_t len)
+		switch (TFW_LC(c)) {
+		case 'c':
+			__FSM_I_JMP(I_Meth_C);
+		case 'd':
+			__FSM_I_JMP(I_Meth_D);
+		case 'g':
+			__FSM_I_JMP(I_Meth_G);
+		case 'h':
+			__FSM_I_JMP(I_Meth_H);
+		case 'l':
+			__FSM_I_JMP(I_Meth_L);
+		case 'm':
+			__FSM_I_JMP(I_Meth_M);
+		case 'o':
+			__FSM_I_JMP(I_Meth_O);
+		case 'p':
+			__FSM_I_JMP(I_Meth_P);
+		case 't':
+			__FSM_I_JMP(I_Meth_T);
+		case 'u':
+			__FSM_I_JMP(I_Meth_U);
+		}
+		__FSM_I_MOVE(I_Meth_Unknown);
+	}
+
+	__FSM_STATE(I_Meth_C) {
+		TRY_STR_LAMBDA("copy", {
+			req->method_override = TFW_HTTP_METH_COPY;
+		} , I_Meth_C, I_EoT);
+		TRY_STR_INIT();
+		__FSM_I_JMP(I_Meth_Unknown);
+	}
+
+	__FSM_STATE(I_Meth_D) {
+		TRY_STR_LAMBDA("delete", {
+			req->method_override = TFW_HTTP_METH_DELETE;
+		} , I_Meth_D, I_EoT);
+		TRY_STR_INIT();
+		__FSM_I_JMP(I_Meth_Unknown);
+	}
+
+	__FSM_STATE(I_Meth_G) {
+		TRY_STR_LAMBDA("get", {
+			req->method_override = TFW_HTTP_METH_GET;
+		} , I_Meth_G, I_EoT);
+		TRY_STR_INIT();
+		__FSM_I_JMP(I_Meth_Unknown);
+	}
+
+	__FSM_STATE(I_Meth_H) {
+		TRY_STR_LAMBDA("head", {
+			req->method_override = TFW_HTTP_METH_HEAD;
+		} , I_Meth_H, I_EoT);
+		TRY_STR_INIT();
+		__FSM_I_JMP(I_Meth_Unknown);
+	}
+
+	__FSM_STATE(I_Meth_L) {
+		TRY_STR_LAMBDA("lock", {
+			req->method_override = TFW_HTTP_METH_LOCK;
+		} , I_Meth_L, I_EoT);
+		TRY_STR_INIT();
+		__FSM_I_JMP(I_Meth_Unknown);
+	}
+
+	__FSM_STATE(I_Meth_M) {
+		TRY_STR_LAMBDA("mkcol", {
+			req->method_override = TFW_HTTP_METH_MKCOL;
+		} , I_Meth_M, I_EoT);
+		TRY_STR_LAMBDA("move", {
+			req->method_override = TFW_HTTP_METH_MOVE;
+		} , I_Meth_M, I_EoT);
+		TRY_STR_INIT();
+		__FSM_I_JMP(I_Meth_Unknown);
+	}
+
+	__FSM_STATE(I_Meth_O) {
+		TRY_STR_LAMBDA("options", {
+			req->method_override = TFW_HTTP_METH_OPTIONS;
+		} , I_Meth_O, I_EoT);
+		TRY_STR_INIT();
+		__FSM_I_JMP(I_Meth_Unknown);
+	}
+
+	__FSM_STATE(I_Meth_P) {
+		TRY_STR_LAMBDA("patch", {
+			req->method_override = TFW_HTTP_METH_PATCH;
+		} , I_Meth_P, I_EoT);
+		TRY_STR_LAMBDA("post", {
+			req->method_override = TFW_HTTP_METH_POST;
+		} , I_Meth_P, I_EoT);
+		TRY_STR_LAMBDA("propfind", {
+			req->method_override = TFW_HTTP_METH_PROPFIND;
+		} , I_Meth_P, I_EoT);
+		TRY_STR_LAMBDA("proppatch", {
+			req->method_override = TFW_HTTP_METH_PROPPATCH;
+		} , I_Meth_P, I_EoT);
+		TRY_STR_LAMBDA("put", {
+			req->method_override = TFW_HTTP_METH_PUT;
+		} , I_Meth_P, I_EoT);
+		TRY_STR_INIT();
+		__FSM_I_JMP(I_Meth_Unknown);
+	}
+
+	__FSM_STATE(I_Meth_T) {
+		TRY_STR_LAMBDA("trace", {
+			req->method_override = TFW_HTTP_METH_TRACE;
+		} , I_Meth_T, I_EoT);
+		TRY_STR_INIT();
+	}
+
+	__FSM_STATE(I_Meth_U) {
+		TRY_STR_LAMBDA("unlock", {
+			req->method_override = TFW_HTTP_METH_UNLOCK;
+		} , I_Meth_U, I_EoT);
+		TRY_STR_INIT();
+		__FSM_I_JMP(I_Meth_Unknown);
+	}
+
+	__FSM_STATE(I_Meth_Unknown) {
+		__FSM_I_MATCH_MOVE(token, I_Meth_Unknown);
+		req->method_override = _TFW_HTTP_METH_UNKNOWN;
+		c = *(p + __fsm_sz);
+		__FSM_I_MOVE_n(I_EoT, __fsm_sz);
```

Comment:
Looks like we can do `p += __fsm_sz` here and then just fall through to the `I_EoT` state (instead of `__FSM_I_MOVE_n()` using) to avoid of repeating of the check that we had already done in `__FSM_I_MATCH_MOVE()` above.

Reply:
Yes, you're right, fall-trough would have absolutely the same behaviour here. But this particular case is not a hot-path, so I prefered to have explicit `move` statement here.